src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -165,8 +168,16 @@ module Make_update (T : Transaction_snark.Verification.S) = struct
         in
         result
       in
+      let%bind is_first_block =
```

Comment:
Can you write a comment explaining how this works somewhere? I can't seem to unwind everything myself at the moment :P